### PR TITLE
Start the sync worker in the builder

### PR DESCRIPTION
**Pull request by codabrink**
- Start the sync worker in the builder.
- Have the sync worker wait until the identity is ready before starting.

### CODE REVIEW

File: xmtp_mls/src/api/test_utils.rs [hunk 1]
```diff
@@ -176,3 +181,34 @@ mod wasm {
         }
     }
 }
+
+pub async fn wait_for_some<F, T>(f: F) -> Option<T>
+where
+    F: Fn() -> Option<T>,
+{
+    let start = Instant::now();
```
Comment (insipx) [inline xmtp_mls/src/api/test_utils.rs:189]: Try to prefer https://docs.rs/web-time/latest/web_time/ `Instant` fn, so that it works in webassembly too

Comment (insipx) [reply xmtp_mls/src/api/test_utils.rs:189]: #1366 puts all these into a common crate for easier access, but that PR still needs some love to get tests passing

Comment (codabrink) [reply xmtp_mls/src/api/test_utils.rs:189]: Cool, thanks

**Review by insipx — APPROVED** [review]

File: xmtp_mls/src/groups/device_sync.rs [hunk 2]
```diff
@@ -113,55 +113,81 @@ impl RetryableError for DeviceSyncError {
...
+        inbox_id = client.inbox_id(),
+        installation_id = hex::encode(client.installation_public_key()),
+        "starting sync worker"
+    );
+    let receiver = client.local_events.subscribe();
+    let sync_stream = receiver.stream_sync_messages();
+
+    // Wait for the identity to be ready before doing anything
+    while !client.identity().is_ready() {
+        crate::sleep(Duration::from_millis(200)).await;
```
Comment (insipx) [inline xmtp_mls/src/groups/device_sync.rs:163]: can also `tokio::yield_now()` to yield back to the executor

**Pull request merged**